test/Microsoft.AspNetCore.JsonPatch.Test/IntegrationTests/AnonymousObjectIntegrationTest.cs

```diff
@@ -0,0 +1,242 @@
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("IntegerList/4", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            var exception = Assert.Throws<JsonPatchException>(() =>
+            {
+                deserialized.ApplyTo(targetObject);
+            });
+            Assert.Equal(
+                string.Format("The index value provided by path segment '{0}' is out of bounds of the array size.", "4"),
+                exception.Message);
+        }
+
+        [Fact]
+        public void AddToList_AtBeginning()
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("IntegerList/0", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(targetObject);
+
+            Assert.Equal(new List<int>() { 4, 1, 2, 3 }, targetObject.IntegerList);
+        }
+
+        [Fact]
+        public void AddToList_InvalidPositionTooSmall()
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("IntegerList/-1", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            var exception = Assert.Throws<JsonPatchException>(() =>
+            {
+                deserialized.ApplyTo(targetObject);
+            });
+            Assert.Equal(
+                string.Format("The index value provided by path segment '{0}' is out of bounds of the array size.", "-1"),
+                exception.Message);
+        }
+
+        [Fact]
+        public void AddToList_Append()
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("IntegerList/-", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(targetObject);
+
+            Assert.Equal(new List<int>() { 1, 2, 3, 4 }, targetObject.IntegerList);
+        }
```

Comment:
I'm not sure there's much value in having tests that use a strategy like this. 

For instance these are the 'anonymous object' tests, but 5 that are related to the behavior of lists. If anything those should be the 'list tests'. If they are duplicated we can remove them.